src/app/resources/resources-view.component.ts

```diff
@@ -41,27 +43,46 @@ export class ResourcesViewComponent implements OnInit {
   urlPrefix = environment.couchAddress + 'resources/';
 
   ngOnInit() {
-    this.route.paramMap.pipe(switchMap((params: ParamMap) => this.getResource(params.get('id'))))
-      .subscribe(resource => this.resource = resource);
+    this.route.paramMap.pipe(switchMap((params: ParamMap) => this.getResource(params.get('id'), params.get('nationname'))))
+      .subscribe(resource => this.resource = resource );
   }
 
-  getResource(id: string) {
-    return this.couchService.get('resources/' + id)
-      .then((data) => {
-        // openWhichFile is used to label which file to start with for HTML resources
-        const filename = data.openWhichFile || Object.keys(data._attachments)[0];
-        this.mediaType = data.mediaType;
-        this.contentType = data._attachments[filename].content_type;
-        this.resourceSrc = this.urlPrefix + data._id + '/' + filename;
-        if (this.mediaType === 'pdf') {
-          this.pdfSrc = this.sanitizer.bypassSecurityTrustResourceUrl(this.resourceSrc);
-        }
-        if (this.mediaType === 'zip') {
-          this.router.navigate([ '/resources' ]);
-          window.open(this.resourceSrc);
-        }
-        return data;
-      }, (error) => console.log('Error'));
+  getResource(id: string, nationname: string) {
+    if (nationname !== null) {
+      return this.couchService.post(`nations/_find`,
+      { 'selector': { 'name':  this.route.snapshot.paramMap.get('nationname') },
+      'fields': [ 'name', 'nationurl' ] })
+        .then(data => {
+          const nationUrl = data.docs[0].nationurl;
+          if (nationUrl) {
+            this.jsonp.request('http://' + nationUrl + '/resources/' + id + '?include_docs=true&callback=JSONP_CALLBACK')
+            .subscribe(res => {
+              const filename = res['_body'].openWhichFile || Object.keys(res['_body']._attachments)[0];
+              this.mediaType = 'other';
+              this.contentType = res['_body']._attachments[filename].content_type;
+              this.resourceSrc = 'http://' + nationUrl + '/resources/' + id + '/' + filename;

```

Comment:
how does it show? as we need to show content in current planet app from different nation not to take them to other nation